util/encoding/encoding_test.go

```diff
@@ -60,3 +61,36 @@ func TestScalarHexString(t *testing.T) {
 	ErrFatal(err)
 	require.True(t, sc.Equal(s2))
 }
+
+// Tests for error cases
+type MockFailingReader struct {
+	data []byte
+}
+type MockEmptyReader struct {
+	data []byte
+}
+
+func (m *MockFailingReader) Read(p []byte) (n int, err error) {
+	return copy(p, m.data), io.EOF
+}
+func (m *MockEmptyReader) Read(p []byte) (n int, err error) {
+	return 0, nil
+}
+
+func TestReadHexPointError(t *testing.T) {
```

Comment:
I'm going to be annoying on a point that's not common in this codebase, that is a bit of a pet peeve of mine, but which I think is important nonetheless:

This test contains 3 different test cases. When possible, try to have independent unit tests.
- This way, when the test fails, it tells you directly that it's the "reader fails" that fails, you don't have to look up which line this failed at.
- Furthermore, if the 3rd test case passes, you won't be able to tell (from the test results) if everything's grouped in a single test.
- Finally, having separate tests makes them more (trivially) independent of one another, and potentially parallelizable (not that it matters here)